### PR TITLE
Ed25519 ChaCha20 Poly1305

### PR DESCRIPTION
**Pull request by Mischala**
Adddong Ed25519, ChaCha20 and Poly11305 to bring webcrypto-core in line with [RFC7539](https://tools.ietf.org/html/rfc7539)

I plan to implement these downstream in webcrypto-liner, for use in js-jose

### CODE REVIEW

**Comment by microshine** [comment]
@Mischala thank you for you PR and work you did.

I'm not sure about `X25519` algorithm. Current version supports X25519 via Ecdh provider. We need to decide which variant to stay. @rmhrisk what do you think?

- [The ECDH-CURVE25519 Algorithm for WebCrypto](https://www.w3.org/2012/webcrypto/WebCryptoCurve25519/Curve25519-WebCrypto.html)
- [node-webcrypto-p11 EC test with X25519](https://github.com/PeculiarVentures/node-webcrypto-p11/blob/master/test/ec.js#L13)
- X25519 uses the same functions like ECDH but only with namedCureve:X25519. [webcrypto-examples](https://github.com/diafygi/webcrypto-examples#ecdh---derivekey)

File: src/chacha20/crypto.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public static decrypt(algorithm: Algorithm, key: CryptoKey, data: Uint8Array): PromiseLike<ArrayBuffer> {
+        return new Promise((resolve, reject) => {
+            this.checkAlgorithmParams(algorithm);
+            this.checkKey(key, this.ALG_NAME, "secret", "decrypt");
+            resolve(undefined);
+        });
+    }
+
+    public static checkKeyGenParams(alg: AesKeyGenParams) {
+        if (alg.length !== 256) {
```
[the remark this answers is not in the document]
Comment (Mischala) [reply src/chacha20/crypto.ts]: the RFC for ChaCha20 specifies a keylength of 256
https://tools.ietf.org/html/rfc7539#page-10
would this change not allow for a weakened implementation of ChaCha20?

**Comment by Mischala** [comment]
@microshine oh i was not aware X25519 was already implemented.
I am happy to remove that addition if X25519 key generation and derivation is already possible.
no point in including two ways to do the same thing.

**Comment by rmhrisk** [comment]
@microshine please review and merge as appropriate.

**Comment by Mischala** [comment]
Hey, any news on this?
would like to get this merged so I can implement these ciphers downstream.

**Comment by microshine** [comment]
Sorry for delay. Thank you for your update and PR

**Comment by microshine** [comment]
I've published new version of webcrypto-core v0.1.24

**Pull request merged**